Packs/Lansweeper/Integrations/Lansweeper/Lansweeper.py

```diff
@@ -0,0 +1,523 @@
+        outputs_prefix="Lansweeper.IP",
+        outputs_key_field="assetId",
+        outputs=context,
+        readable_output=readable_hr,
+        raw_response=response
+
+    )
+
+
+def lansweeper_mac_hunt_command(client: Client, args: Dict[str, str]) -> CommandResults:
+    """
+    Retrieves the list of assets
+    :type client: ``Client``
+    :param client: Client object to be used.
+
+    :type args: ``Dict[str, str]``
+    :param args: The command arguments provided by the user.
+
+    :return: Standard command result or no records found message.
+    :rtype: ``CommandResults``
+    """
+    total_records = []
+    site_list = prepare_site_list(client, args)
+    limit = arg_to_number(args.get('limit', 50))
+    mac_list = argToList(args['mac_address'])
+
+    if not mac_list:
+        raise ValueError(MESSAGES["REQUIRED_ARGUMENT"].format('mac_address'))
+
+    if limit <= 0 or limit > 500:  # type:ignore
+        raise ValueError(MESSAGES["INVALID_LIMIT"].format(limit))
+
+    mac_condition = ""
+    for mac in mac_list:
+        if is_mac_address(mac):
+            mac_condition += ("""{operator: EQUAL,path: "assetBasicInfo.mac",value: "%s"},""" % mac)
+    if not mac_condition:
+        raise ValueError(MESSAGES["INVALID_MAC"])
+
+    for site in site_list:
+        query = prepare_query(site.get('id'), mac_condition[:-1])
+        response = client.asset_list(query)
+        records = response.get("data", {}).get("site", {}).get("assetResources", {}).get("items", [])
+        for record in records:
+            record['assetId'] = record.pop('_id')
+            record['siteId'] = site.get('id')
+            record['siteName'] = site.get('name')
+            total_records.append(remove_empty_elements(record))
+
+    context = total_records[:limit]
+    readable_hr = prepare_hr_for_asset(context)
+    return CommandResults(
+        outputs_prefix="Lansweeper.Mac",
+        outputs_key_field="assetId",
+        outputs=context,
+        readable_output=readable_hr,
+        raw_response=response
+
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises:
+     exceptions if something goes wrong.
+
+    Args:
+        client: client to use
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    client.site_list()
+    return 'ok'
+
+
+def main():
+    """main function, parses params and runs command functions"""
+
+    command = demisto.command()
+    demisto.debug(f'[Lansweeper] Command being called is {command}')
+
+    params = demisto.params()
+
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+
+    identity_code = params.get("identity_code")
+    if creds_changed(get_integration_context(), identity_code):
+        update_context(identity_code)
```

Comment:
Can you please explain what you tried to achieve here? Why should we save the identity_code to the integration context?

Reply:
For the configured identity_code, we are here calling an endpoint to retrieve the authorized sites and storing the details of the authorized sites in the integration context. These sites will be used for fetching the asset details in other commands.

If the configured identity code is not modified, then instead of calling endpoint we'll retrieve it from integration context thereby reducing number of API calls.

So we're just checking to see if the identity_code has been changed or not. If this is the case, update the integration context accordingly.